clusters/clusters_api.go

```diff
@@ -531,6 +537,24 @@ func (a ClustersAPI) Create(cluster Cluster) (info ClusterInfo, err error) {
 	return
 }
 
+// Resize api can only be used when the cluster is in Running State
+func (a ClustersAPI) Resize(resizeRequest ResizeRequest) (info ClusterInfo, err error) {
+	info, err = a.Get(resizeRequest.ClusterID)
+	if err != nil {
+		return info, err
+	}
+	if info.State != ClusterStateRunning {
+		return info, fmt.Errorf("Resize API can only be called on clusters in RUNNING state. Cluster %v is in %v state", info.ClusterID, info.State)
+	}
+
+	err = a.client.Post(a.context, "/clusters/resize", resizeRequest, &info)
+	if err != nil {
+		return info, err
```

Comment:
```suggestion
		return info, fmt.Errorf("resize: %w", err)
```

in go we have to wrap error messages, so that for users it's easier to understand the error

Reply:
done